### PR TITLE
[SPIKE] Transactional session

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

File: src/NServiceBus.TransactionalSession.AcceptanceTests/TestingInfrastructure/TransactionSessionDefaultServer.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    public static Action<EndpointConfiguration> ConfigurePersistence { get; set; } =
+        _ => throw new NotImplementedException();
+
+    public override Task<EndpointConfiguration> GetConfiguration(RunDescriptor runDescriptor,
+        EndpointCustomizationConfiguration endpointConfiguration,
+        Action<EndpointConfiguration> configurationBuilderCustomization)
+    {
+        endpointConfiguration.TypesToInclude.Add(typeof(CaptureBuilderFeature));
+        endpointConfiguration.TypesToInclude.Add(typeof(CosmosDbTransactionalSessionFeature));
```
Comment (danielmarbach) [inline src/NServiceBus.TransactionalSession.AcceptanceTests/TestingInfrastructure/TransactionSessionDefaultServer.cs:18]: That's the thing I don't understand why I need to do this currently

File: src/NServiceBus.Persistence.CosmosDB/NServiceBus.Persistence.CosmosDB.csproj [hunk 2]
```diff
@@ -5,6 +5,7 @@
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>..\NServiceBus.snk</AssemblyOriginatorKeyFile>
     <LangVersion>8.0</LangVersion>
+    <RootNamespace>NServiceBus.TransactionalSession</RootNamespace>
```
Comment (timbussmann) [inline src/NServiceBus.Persistence.CosmosDB/NServiceBus.Persistence.CosmosDB.csproj:8]: that looks off?

File: src/NServiceBus.TransactionalSession.AcceptanceTests/TestingInfrastructure/TransactionSessionDefaultServer.cs (hunk 1 above)
Comment (timbussmann) [reply src/NServiceBus.TransactionalSession.AcceptanceTests/TestingInfrastructure/TransactionSessionDefaultServer.cs:18]: all types from the acceptance test project itself are excluded from scanning by default (https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/415/files#diff-49ec7e99da19569a2afdbe8174060351275338bfc3419aa4dc6cfdcd4da77efdR22) and `Features` can only be activated if they are scanned, explicitly enabling them via the `EnableFeature` API doesn't suffice (raised an issue for that already)

**Review by timbussmann — COMMENTED** [review]
LGTM

File: src/NServiceBus.Persistence.CosmosDB/NServiceBus.Persistence.CosmosDB.csproj (hunk 2 above)
Comment (danielmarbach) [reply src/NServiceBus.Persistence.CosmosDB/NServiceBus.Persistence.CosmosDB.csproj:8]: The idea was to put everything into the same feature namespace of transactional session

**Pull request closed without merging**